api/storage/client.go

```diff
@@ -82,3 +82,15 @@ func (c *Client) CreatePool(pname, ptype string, pconfig map[string]interface{})
 	}
 	return c.facade.FacadeCall("CreatePool", args, nil)
 }
+
+// ListVolumes lists volumes according to a given filter.
+// If no filter was provided, this will return a list
+// of all volumes.
+func (c *Client) ListVolumes(machines []string) ([]params.StorageDisk, error) {
```

Comment:
I'd like to cut down on the terminology, which is why I winced a bit when the suggestion of Disk came up in the standup. We can defer this, but I think we may want to just auto-create Volumes from discovered block devices if there are no matching provider-created Volumes.